docs/ldml/tr35-keyboards.md

```diff
@@ -2030,6 +2039,221 @@ Footnotes:
+
+See the examples section below for all valid attribute combinations. In reviewing the attributes, see also [`<key>`](#Element_key).
+
+If a key is not found, or a particular gesture has no definition, the output should be behave as if the user performed such an action.  For example, an unspecified flick would result in no output.
+
+_Attribute:_ `to`
+
+This attribute specifies a key by means of the key’s `to` attribute.
+
+It is an error to use this attribute with a `to` value that matches more than one key.  Such keys are not invalid, however, they must be selected here by using the `key` attribute and specifying their key id.
+
+_Attribute:_ `key`
+
+This attribute specifies a key by means of the key’s `id` attribute.
+
+_Attribute:_ `flick`
+
+This attribute specifies a flick gesture to be performed on the specified key instead of a keypress, such as `e` or `nw se`. This value corresponds to the `directions` attribute of the [`<flick>`](#Element_flicks) element.
+
+_Attribute:_ `longPress`
+
+This attribute specifies that a long press gesture should be performed on the specified key instead of a keypress. For example, `longPress="2"` indicates that the second character in a longpress series should be chosen. `longPress="0"` indicates that the `longPressDefault` value, if any, should be chosen. This corresponds to `longPress` and `longPressDefault` on [`<key>`](#Element_key).
+
+_Attribute:_ `tapCount`
+
+This attribute specifies that a multi-tap gesture should be performed on the specified key instead of a keypress. For example, `tapCount="3"` indicates that the key should be tapped three times in rapid succession. This corresponds to `multiTap` on [`<key>`](#Element_key).
+
+Note that `tapCount="1"` is valid, but represents an ordinary keypress.
+
+_Attribute:_ `result` (required)
+
+This element specifies the expected resultant text in a document after processing this event and all prior events, and including any `preContext` text.  This text may be escaped with `\u` notation, see [Escaping](#Escaping).
+
+Examples:
+
+```xml
+<test name="spec-sample">
+    <startContext to="abc\u0022"/>
+    <!-- simple, key specified by to -->
+    <event to="s"                   result="abc\u0022s"/>
+    <!-- simple, key specified by id -->
+    <event key="doublequote"        result="abc\u0022s\u0022"/>
+    <!-- flick -->
+    <event key="s" flick="nw se"    result="abc\u0022s\u0022•"/>
+    <!-- longPress -->
+    <event key="e" longPress="1"    result="abc\u0022s\u0022•é"/>
+    <!-- multiTap -->
+    <event key="E" tapCount="2"     result="abc\u0022s\u0022•éÈ"/>
+    <!-- flick, with to -->
+    <event to="s" flick="nw se"     result="abc\u0022s\u0022•éÈ•"/>
+    <!-- longPress, with to -->
+    <event to="e" longPress="1"     result="abc\u0022s\u0022•éÈ•é"/>
+    <!-- multiTap, with to -->
+    <event tongue="E" tapCount="2"  result="abc\u0022s\u0022•éÈ•éÈ"/>
```

Comment:
I think we should instead make `to` just an output character, so these can't be combined with `flick`, `longPress` and `tapCount`. It'll be very helpful for testing transforms.

Reply:
Perhaps we should use a different element rather than `event`? Rename `event` to `<keystroke key="x" flick... />` and have a separate element `<emit to="..." result="..." />`?

The linter for the test can verify if `x` in `<emit to="x">` can be generated by any key or sub-key, to close the circle.

Reply:
> different element rather than event

Yeah, when I wrote the paragraph about the combinatoric rules I was kind of getting there. 

Sgtm

Reply:
ptal